fetchmail_inbox/models/fetchmail_inbox.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2014-2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, models
+
+
+class FetchmailInbox(models.Model):
+    _inherit = 'mail.thread'
+    _name = 'fetchmail.inbox'
+    _description = 'Fetchmail inbox'
+
+    @api.multi
+    def name_get(self):
+        mails = self.env['mail.message'].search([
+            ('model', '=', self._name),
+            ('res_id', 'in', self.ids),
+        ])
```

Comment:
here we can expect more then we like I guess, so I think a `dict((this.id, this) for this in mails)` once here is warranted to avoid the filtering later

Reply:
Thanks for the review! Now using something different altogether, hope you like that. Added a test for this too.